### DISCUSSION TITLE
Putting Corefreqd in the root cgroup

### DISCUSSION THREAD

**Discussion by gel-crabs** [cyring/CoreFreq#379] [General]
Hey, I noticed in the troubleshooting section it is recommended to put corefreqd in the root cgroup; I did a lot of research finding out how to do this with systemd, and I found out how; in corefreqd.service, under [Service], put Slice=-.slice, like so:

```
# CoreFreq
# Copyright (C) 2015-2022 CYRIL COURTIAT
# Licenses: GPL2

[Unit]
Description=CoreFreq Daemon

[Service]
Type=simple
ExecStart=corefreqd -q
ExecStop=/bin/kill -QUIT $MAINPID
RemainAfterExit=no
SuccessExitStatus=SIGQUIT SIGUSR1 SIGTERM
Slice=-.slice

[Install]
WantedBy=multi-user.target
```
systemd-cgls then shows it under the root cgroup after rebooting. Is this something that should be added to the service file?

**Comment by cyring** [comment]
Nice, thanks for the tip.

For your testings, it is now available in the `develop` branch at commit 0e95b1161d2be5cbf28f0752421be7fc21ea74bc

**Comment by gel-crabs** [comment]
Just did a couple geekbenches, performance has increased

Without cgroup modification: https://browser.geekbench.com/v5/cpu/19224299
With cgroup modification: https://browser.geekbench.com/v5/cpu/19224374

**Reply by cyring** [reply]
I'm not familiar with container but do you have an idea of the root cause ?

**Reply by gel-crabs** [reply]
Does corefreqd change the P-states, idle states, etc. or is it just the kernel module that does that?

**Reply by cyring** [reply]
P-states and C-states are changed if manually requested/selected from the UI. 
They can however be initialized with a few features, like:
* activating Turbo with Intel NEHALEM 
* registering CPU-Idle
* activating Intel HWP or AMD CPPC

I suggest you show me `corefreq-cli` outputs before/after increasing performance.

**Reply by gel-crabs** [reply]
What command would you like me to run to get the needed output?

**Reply by cyring** [reply]
> What command would you like me to run to get the needed output?

> Does corefreqd change the P-states, idle states, etc. or is it just the kernel module that does that?

Forget the claimed outputs because I guarantee nothing, unless a bug, is touched against performance.
Regards

**Reply by gel-crabs** [reply]
That's fair, I've since gathered that Geekbench scores tend to vary quite a bit anyway
